src/main.js

```diff
@@ -3,13 +3,22 @@ import {
   isClassicFacebook,
 } from "./classic";
 import { setupPageObserver as setupFB5PageObserver, isFB5 } from "./fb5";
+import {
+  setupPageObserver as setupFB5WatchObserver,
+  isFBWatch,
+} from "./fb_watch";
 
 if (isClassicFacebook()) {
   // Old Facebook design
   setupClassicPageObserver();
 } else if (isFB5()) {
   // if it's FB5 design
-  setupFB5PageObserver();
+  if (isFBWatch()) {
+    // if it's facebook.com/watch
+    setupFB5WatchObserver();
+  } else {
+    setupFB5PageObserver();
+  }
```

Comment:
If it is setup like this, will our extension blocks ads after a user navigates to homepage with news feed?

Is it better to add this functionality to block to `/watch` in `fb5.js` instead?

Reply:
You're correct i didn't test it precisely.